magda-web-client/src/Components/Dataset/Add/DatasetAddCommon.ts

```diff
@@ -1234,20 +1230,59 @@ async function convertStateToDistributionRecords(
     return distributionRecords;
 }
 
+/**
+ * Update ckan export aspect status acccording to UI status.
+ * We use JSON Patch request to avoid edging cases
+ *
+ * @param {string} datasetId
+ * @param {State} state
+ */
+async function updateCkanExportStatus(datasetId: string, state: State) {
+    /**
+     * Please note: any frontend code should ONLY (and should ONLY NEED to) change those three fields:
+     * - status
+     * - exportRequired
+     * Those two fields should ONLY been updated via patch request. Otherwise, other runtime fields may lose value.
+     *
+     * Other fields are all runtime status fields that are only allowed to be altered by minions.
+     * Any attempts to update those from frontend will more or less create edging cases.
```

Comment:
Once again, it'd be good to say what the edge case is.

Reply:
Added comment above